rapids-cmake/cpm/gbench.cmake

```diff
@@ -0,0 +1,75 @@
+#=============================================================================
+# Copyright (c) 2022, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#=============================================================================
+include_guard(GLOBAL)
+
+#[=======================================================================[.rst:
+rapids_cpm_gbench
+-----------------
+
+.. versionadded:: v21.12.00
+
+Allow projects to find or build `Google Benchmark` via `CPM` with built-in
+tracking of these dependencies for correct export support.
+
+Uses the version of GBench :ref:`specified in the version file <cpm_versions>` for consistency
+across all RAPIDS projects.
+
+.. code-block:: cmake
+
+  rapids_cpm_gbench( [BUILD_EXPORT_SET <export-name>]
+                     [INSTALL_EXPORT_SET <export-name>]
+                     [<CPM_ARGS> ...])
+
+.. |PKG_NAME| replace:: GBench
+.. include:: common_package_args.txt
+
+Result Targets
+^^^^^^^^^^^^^^
+  benchmark::benchmark targets will be created
```

Comment:
Looks like `gbench` also has a benchmark::benchmark_main target so lets expose that as well.